### DISCUSSION TITLE
Allow "replay" of alerts on past data

### DISCUSSION THREAD

**Discussion by pepjo** [grafana/grafana#52693] [Alerting]
# Problem
When dealing with some devices that have limited connectivity (such as IoT), sometimes data sources can get back-filled with data when it becomes available. This happens because those devices will have periods when they are "offline" but continue to record data and send all the data in batches when they are "online". In such cases, it is great to have the ability to scan past data to see if an alert would have been triggered and if it is still triggered.

There are two scenarios that can be improved from the current implementation:
- When an alert would have been active when the device is "offline" and is resolved before the device is "online" and hence the data is sent to the data source. In this case, the current alerting system never detects that there was a problem.
- In addition, when alerts start in the "offline" period but are still active when the device becomes "online", it is useful to show in the alert notification when the problem actually started. With the current system, the problem is actually detected but the start time is equal to the time the data becomes available, not the time the problem started.

# Proposed solution
As far as I know, this could be solved by adding a "replay" evaluation endpoint to request certain alert rules to be re-evaluated for a defined amount of time. For example, "re-evaluate alert rule XX between 10:00am and 12:00am today". The rule would still follow the evaluation interval defined, so, if that is set to 10m, this example would evaluate the alert a total of 120 times to cover the 2h gap.

This is something that is implemented in VictoriaMetrics: https://github.com/grafana/grafana/discussions/37012 , albeit not with an endpoint but with a CLI command.

# Miscellaneous
Please let me know if this is already been discussed, the only issue I could find that was possibly related is: https://github.com/grafana/grafana/discussions/37012, but the proposed solution would not solve the problem exposed.

Finally, I do work with golang and typescript on a day to day basis and I would happy to contribute if this is considered an interesting feature.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
